services/src/main/java/org/keycloak/services/clienttype/client/TypeAwareClientModelDelegate.java

```diff
@@ -216,50 +216,23 @@ public void removeRedirectUri(String redirectUri) {
 
     @Override
     public void setAttribute(String name, String value) {
-        TypedClientExtendedAttribute attribute = TypedClientExtendedAttribute.getAttributesByName().get(name);
-        if (attribute != null) {
-            attribute.setClientAttribute(clientType, value, (newValue) -> super.setAttribute(name, newValue), String.class);
-        } else {
+        if (clientType.isApplicable(name)) {
```

Comment:
In case that `value` is defined on clientType, we don't want to allow to set attribute IMO? Or is it handled elsewhere?